src/input/mod.rs

```diff
@@ -213,6 +213,25 @@ impl State {
                     self.common.idle_notifier_state.notify_activity(&seat);
 
                     let keycode = event.key_code();
+
+                    const POWER_OFF: u32 = 116;
+                    // if power key is pressed, just use the system action for power off
+                    // this is a workaround for the fact that the power key is not
+                    // available in the keymap, so we cannot use the keymap to determine
+                    // if the key is pressed
+                    if keycode.raw() == POWER_OFF + X11_KEYCODE_OFFSET {
+                        self.handle_action(
+                            Action::Shortcut(shortcuts::Action::System(
+                                shortcuts::action::System::PowerOff,
+                            )),
+                            &seat,
+                            SERIAL_COUNTER.next_serial(),
```

Comment:
Also given that we continue with the rest of the function (which is fine to make sure the keyboard state is correct), we don't want to increase the serial twice.

So probably just move this block after the variable setup and also re-use `state` and `time`?